src/mmds/src/lib.rs

```diff
@@ -59,14 +74,7 @@ pub fn json_patch(target: &mut Value, patch: &Value) {
     }
 }
 
-fn build_response(http_version: Version, status_code: StatusCode, body: Body) -> Response {
-    let mut response = Response::new(http_version, status_code);
-    response.set_body(body);
-    response
-}
-
-pub fn parse_request(request_bytes: &[u8]) -> Response {
-    let request = Request::try_from(request_bytes);
+pub fn callback(request: Result<Request, RequestError>) -> Response {
```

Comment:
+1 on renaming `callback()`. From what I see, this function is used in `parse_request()`, maybe it would make sense to rename that function to `parse_request_bytes()` and this one to `convert_to_response()`.